gdal/swig/python/gdal-utils/osgeo_utils/auxiliary/util.py

```diff
@@ -154,3 +222,95 @@ def _open_ds(
         open_options = ["{}={}".format(k, v) for k, v in open_options.items()]
 
         return gdal.OpenEx(str(filename), access_mode, open_options=open_options)
+
+
+def get_data_type(data_type: Optional[DataTypeOrStr]):
+    if data_type is None:
+        return None
+    if isinstance(data_type, str):
+        return gdal.GetDataTypeByName(data_type)
+    else:
+        return data_type
+
+
+def get_raster_bands(ds: gdal.Dataset) -> Iterator[gdal.Band]:
+    return (ds.GetRasterBand(i + 1) for i in range(ds.RasterCount))
+
+
+def get_band_types(filename_or_ds: PathOrDS):
+    with OpenDS(filename_or_ds) as ds:
+        return [band.DataType for band in get_raster_bands(ds)]
+
+
+def get_band_minimum(band: gdal.Band):
+    ret = band.GetMinimum()
+    if ret is None:
+        band.ComputeStatistics(0)
+    return band.GetMinimum()
+
+
+def get_raster_band(filename_or_ds: PathOrDS, bnd_index: int = 1, ovr_index: Optional[int] = None):
+    with OpenDS(filename_or_ds) as ds:
+        bnd = ds.GetRasterBand(bnd_index)
+        if ovr_index is not None:
+            bnd = bnd.GetOverview(ovr_index)
+        return bnd
+
+
+def get_raster_minimum(filename_or_ds: PathOrDS, bnd_index: Optional[int] = 1):
+    with OpenDS(filename_or_ds) as ds:
+        if bnd_index is None:
+            return min(get_band_minimum(bnd) for bnd in get_raster_bands(ds))
+        else:
+            bnd = ds.GetRasterBand(bnd_index)
+            return get_band_minimum(bnd)
+
+
+def get_raster_min_max(filename_or_ds: PathOrDS, bnd_index: int = 1):
+    with OpenDS(filename_or_ds) as ds:
+        bnd = ds.GetRasterBand(bnd_index)
+        bnd.ComputeStatistics(0)
+        min_val = bnd.GetMinimum()
+        max_val = bnd.GetMaximum()
+        return min_val, max_val
+
+
+def get_nodatavalue(filename_or_ds: PathOrDS):
+    with OpenDS(filename_or_ds) as ds:
+        band = next(get_raster_bands(ds))
+        return band.GetNoDataValue()
+
+
+def unset_nodatavalue(filename_or_ds: PathOrDS):
+    with OpenDS(filename_or_ds, access_mode=gdal.GA_Update) as ds:
+        for b in get_raster_bands(ds):
+            b.DeleteNoDataValue()
+
+
+def get_image_structure_metadata(filename_or_ds: PathOrDS, key: str, default=None):
+    key = key.strip()
+    if not key.endswith("="):
+        key = key + "="
+    with OpenDS(filename_or_ds) as ds:
+        metadata = ds.GetMetadata_List("IMAGE_STRUCTURE")
+        if metadata is None:
+            return default
+        for metadata_item in metadata:
+            if metadata_item.startswith(key):
+                return metadata_item[len(key):]
+        return default
+
+
+def get_big_tiff(big_tiff):
```

Comment:
```suggestion
def get_bigtiff_creation_option_value(big_tiff):
```
As far as I know this method can only be called with str or None. Not bool

Reply:
I'm not I understand your comment, but I've modified it for your other remarks, so it might be OK now.

Reply:
Are `'True'`/`'False'` are valid inputs for big_tiff and all the other inputs which accept `'Yes'`/`'No'` ?

Reply:
Any creation option that declares in the XML  type="boolean" accepts yes/true/on and no/false/off (case insensitive) : https://github.com/OSGeo/gdal/blob/362541e961cf8cab046dfbae1bde8ce559b4cb40/gdal/gcore/gdaldriver.cpp#L1973

Reply:
> Any creation option that declares in the XML type="boolean" accepts yes/true/on and no/false/off (case insensitive) :
> 
> https://github.com/OSGeo/gdal/blob/362541e961cf8cab046dfbae1bde8ce559b4cb40/gdal/gcore/gdaldriver.cpp#L1973

Thanks for the reference. I've amended the implementation of `is_true()` to be compatible with that. 
It will raise an exception if an unexpected input is provided.